homeassistant/components/sensor/bom.py

```diff
@@ -66,25 +74,53 @@
     'wind_spd_kt': ['Wind Direction kt', 'kt']
 }
 
+
+def validate_station(station):
+    """Check that the station ID is well-formed."""
+    if station is None:
+        return
+    station = station.replace('.shtml', '')
+    if not re.fullmatch(r'ID[DNQTVW]60801\.\d\d\d\d\d', station):
+        raise vol.error.Invalid('Malformed station ID')
+    return station
+
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_ZONE_ID): cv.string,
-    vol.Required(CONF_WMO_ID): cv.string,
+    vol.Optional(CONF_ZONE_ID): cv.string,
+    vol.Optional(CONF_WMO_ID): cv.string,
     vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_STATION): validate_station,
```

Comment:
Hmm.  I've put `CONF_ZONE_ID` and `CONF_WMO_ID` in an inclusive group, so it's valid to provide both or neither but not one - this does simplify things and is more correct than optional for both.

However I can't work out how to express 'A xor (B and C)' - mixing inclusive and exclusive seems doomed to false-positives or false-negatives in every combination I've tried.  So I'm just logging a warning in `platform_setup`, and removing the documentation of the old style should avoid new users of it.